bug: logfwd-output tests do not compile because StdoutSink is not an OutputSink

## Summary

The current `origin/master` tip does not pass `cargo test -p logfwd-output` because a test in `crates/logfwd-output/src/lib.rs` tries to put `StdoutSink` into a `Vec<Box<dyn OutputSink>>`, but `StdoutSink` does not implement `OutputSink`.

## Reproduction

From a clean checkout at `df59f6bd143bc737a2d198c8b381bd3321114a28`:

```bash
git fetch origin
git switch --detach origin/master
cargo test -p logfwd-output
```

## Actual behavior

The crate fails to compile its tests with `E0277`:

```text
error[E0277]: the trait bound `stdout::StdoutSink: OutputSink` is not satisfied
   --> crates/logfwd-output/src/lib.rs:926:47
926 |         let mut fanout = FanoutSink::new(vec![Box::new(fanout_s1), Box::new(fanout_s2)]);
    |                                               ^^^^^^^^^^^^^^^^^^^ unsatisfied trait bound
...
    = note: required for the cast from `Box<stdout::StdoutSink>` to `Box<dyn OutputSink>`
```

## Expected behavior

`cargo test -p logfwd-output` should compile and run successfully on the default branch.

## Notes

This also breaks the workspace `just test` flow, since test compilation stops in `logfwd-output` before the suite can run.

--- a/crates/logfwd-output/src/lib.rs
+++ b/crates/logfwd-output/src/lib.rs
@@ -920,21 +920,31 @@ mod tests {
         assert_eq!(out1, out2);
         assert!(!out1.is_empty());
 
-        // Also test FanoutSink trait dispatch works.
-        let fanout_s1 = StdoutSink::new(
-            "f1".to_string(),
-            StdoutFormat::Json,
-            Arc::new(ComponentStats::new()),
-        );
-        let fanout_s2 = StdoutSink::new(
-            "f2".to_string(),
-            StdoutFormat::Json,
-            Arc::new(ComponentStats::new()),
-        );
-        let mut fanout = FanoutSink::new(vec![Box::new(fanout_s1), Box::new(fanout_s2)]);
-        // send_batch writes to real stdout, but should not error.
-        let result = fanout.send_batch(&batch, &meta);
-        assert!(result.is_ok());
+        // Test FanoutSink dispatch with sync OutputSink impls.
+        // StdoutSink was migrated to async Sink, so use a minimal NoOp.
+        {
+            struct NoOpSink(&'static str);
+            #[allow(deprecated)]
+            impl OutputSink for NoOpSink {
+                fn send_batch(&mut self, _: &RecordBatch, _: &BatchMetadata) -> io::Result<()> {
+                    Ok(())
+                }
+                fn flush(&mut self) -> io::Result<()> {
+                    Ok(())
+                }
+                fn name(&self) -> &str {
+                    self.0
+                }
+            }
+            #[allow(deprecated)]
+            let mut fanout = FanoutSink::new(vec![
+                Box::new(NoOpSink("f1")) as Box<dyn OutputSink>,
+                Box::new(NoOpSink("f2")),
+            ]);
+            #[allow(deprecated)]
+            let result = fanout.send_batch(&batch, &meta);
+            assert!(result.is_ok());
+        }
     }
 
     struct AlwaysFailSink {
@@ -1147,7 +1157,7 @@ mod tests {
     }
 
     #[test]
-    fn test_build_output_sink_stdout() {
+    fn test_build_sink_factory_stdout() {
         let cfg = OutputConfig {
             name: Some("test".to_string()),
             output_type: OutputType::Stdout,
@@ -1160,8 +1170,9 @@ mod tests {
             index: None,
             auth: None,
         };
-        let sink = build_output_sink("test", &cfg, Arc::new(ComponentStats::new())).unwrap();
-        assert_eq!(sink.name(), "test");
+        // StdoutSink uses the async pipeline — must use build_sink_factory.
+        let factory = build_sink_factory("test", &cfg, Arc::new(ComponentStats::new())).unwrap();
+        assert_eq!(factory.name(), "test");
     }
 
     #[test]

